### PR TITLE
Correctly handle pass-through "composites" in the Fn Api Runner.

### PR DESCRIPTION
**Pull request by robertwb**
Also cleaned up some nearby tests.

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R:@udim

File: sdks/python/apache_beam/runners/portability/fn_api_runner.py [hunk 1]
```diff
@@ -773,8 +773,12 @@ def process(stage):
     def leaf_transforms(root_ids):
       for root_id in root_ids:
         root = pipeline_proto.components.transforms[root_id]
-        if root.spec.urn in known_composites or not root.subtransforms:
+        if root.spec.urn in known_composites:
           yield root_id
+        elif not root.subtransforms:
+          # Make sure its outputs that are not a subset of its inputs.
```
Comment (udim) [inline sdks/python/apache_beam/runners/portability/fn_api_runner.py]: s/that//

**Comment by udim** [comment]
retest this please

**Comment by robertwb** [comment]
retest this please

**Pull request merged**